src/lib/RuleFilter.ts

```diff
@@ -1,18 +1,216 @@
-export enum FilterType {
-	RULENAME,
-	CATEGORY,
-	RULESET,
-	LANGUAGE,
-	SOURCEPACKAGE,
-	ENGINE
-}
-
-export class RuleFilter {
-    readonly filterType: FilterType;
-    readonly filterValues: ReadonlyArray<string>;
-
-    constructor(filterType: FilterType, filterValues: string[]) {
-        this.filterType = filterType;
-        this.filterValues = filterValues.map(v => v.trim());
-    }
+import { SfdxError } from '@salesforce/core';
```

Comment:
The RuleFilters are now Data + Behavior instead of just Data. This allows for easier testing and less duplication of code for the classes interacting with the filters.